app/controllers/posts_controller.rb

```diff
@@ -45,20 +47,17 @@ def owed
 
   def unread
     @started = (params[:started] == 'true') || (params[:started].nil? && current_user.unread_opened)
-    @posts = Post.joins("LEFT JOIN post_views ON post_views.post_id = posts.id AND post_views.user_id = #{current_user.id}")
-    @posts = @posts.joins("LEFT JOIN board_views on board_views.board_id = posts.board_id AND board_views.user_id = #{current_user.id}")
+    @posts = Post.not_ignored_by(current_user)
     @posts = @posts.where.not(post_views: { read_at: nil }) if @started
 
-    # post view does not exist and (board view does not exist or post has updated since non-ignored board view read_at)
+    # post view does not exist and (board view does not exist or post has updated since board view read_at)
     no_post_view = @posts.where(post_views: { user_id: nil })
     updated_since_board_read = no_post_view.where(board_views: { read_at: nil })
       .or(no_post_view.where("date_trunc('second', board_views.read_at) < date_trunc('second', posts.tagged_at)"))
-      .where(board_views: { ignored: false })
     no_post_view = no_post_view.where(board_views: { user_id: nil }).or(updated_since_board_read)
 
-    # post view exists and post has updated since non-ignored post view read_at and (board view does not exist or is not ignored)
-    with_post_view = @posts.where(post_views: { ignored: false }) # non-existant post-views will return nil here
-    with_post_view = with_post_view.where(board_views: { user_id: nil }).or(with_post_view.where(board_views: { ignored: false }))
+    # post view exists and post has updated since post view read_at
+    with_post_view = @posts.where.not({ user_id: nil })
```

Comment:
I'm not sure I understand this one - posts where the post has an author? 🤔

Reply:
good... question. it's been a while since I wrote this and I do *not* remember.

Reply:
todo: need to look into this in more detail

Reply:
okay so I *think* what this is *trying* to do is select posts that have a post view and should be `@posts.where.not(post_views: { user_id: nil })`. I'm not sure that actually does the right thing but I think that's what happened there.

Reply:
Aha, that makes sense! Nice